docs: Prevent console access (#7398)
zitadel/zitadel: docs/docs/guides/solution-scenarios/disable-console.mdx

---
title: Prevent users from accessing ZITADEL console

> sidebar_label: Disable ZITADEL Console

Reviewer: ```suggestion
sidebar_label: Limit access to console
```

Reviewer: We're not really disabling console

---

> ZITADEL does include the ZITADEL Console, which allows you to manage all ZITADEL resources as well as end users to manage their profile information.

Reviewer: ```suggestion
ZITADEL includes a console that allows Managers to configure all resources. All uses, including end-users, by default, view and manage their profile information.
```

In some use cases you want to prevent users from accessing the console, this could be:
- User management is integrated into your own app
- Users login via SSO and should not be able to change their data in the Console
- Only managers should be able to access Console to manage their users

> At the moment it is not possible to simply disable the Console but with a workaround you can prevent users from accessing the Console.

Reviewer: ```suggestion
At the moment it is not possible to simply disable the Console but with the configuration described in this guide you can prevent users from accessing the Console.
```

## Self-hosted

If you are running ZITADEL self-hosted we recommend to restrict the access to ZITADEL Console via WAF/reverse-proxy for Non-Manager users

## ZITADEL Cloud (and self-hosted)

### Default redirect url

> The Default Redirect URI will be used, if a user calls the login page directly. More specifically, typically a client will initiate login with an auth request.

Reviewer: ```suggestion
The Default Redirect URI will be used, if a user calls the login page directly. More specifically, typically a client will initiate login with an [auth request](https://zitadel.com/docs/apis/openidoauth/authrequest).
```

Reviewer: Should be actually linked and explained in https://github.com/zitadel/zitadel/issues/7399

The auth request contains a client-id and a redirect uri, that must match the configuration in ZITADEL.
If there is no auth request, users will be redirected to the Default Redirect URI, which is by default https://<custom_domain>/ui/console/

Change default redirect url of instance: https://<custom_domain>/ui/console/settings?id=login

![Login Policy Advanced Setting: Default Redirect URL](/img/guides/solution-scenarios/console-default-redirect.png)

### Restricting Console in default-project

> With this workaround, you will limit users from accessing the default-project, if they are not explicitly granted to use this project.

Reviewer: Can we explain default-project? Or even better add that as "Concept" for reference?

Author: added

To avoid accidental lock-out, the default project hides the checkbox "Check Project on Authentication" in the Console:
![Default Project](/img/guides/solution-scenarios/console-default-project.png)

You need to use the [update project API call](/docs/apis/resources/mgmt/management-service-update-project) to set the configuration on the project.

First you need a user with manager permissions to change the project settings in that organization.
This means either you add the manager to the organization or you use a manager on the instance level with "IAM-OWNER" permissions.
After that create a Personal Access Token (PAT) for the manager.

More detailed information about creating a PAT and manager roles you can find [here](/docs/guides/integrate/pat)

Then you have to send the following request:
```bash
curl -L -X PUT "https://$CUSTOM_DOMAIN/management/v1/projects/$PROJECT_ID" \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-H "Authorization: Bearer $PAT" \
--data-raw '{
  "name": "ZITADEL",
  "projectRoleAssertion": false,
  "projectRoleCheck": false,
  "hasProjectCheck": true,
  "privateLabelingSetting": "PRIVATE_LABELING_SETTING_UNSPECIFIED"
}'
```

Where

> - CUSTOM_DOMAIN is your domain, f.e. b2b-demo-rbxajm.zitadel.cloud

Reviewer: ```suggestion
- CUSTOM_DOMAIN is your domain, f.e. mydemo.zitadel.cloud
```
